src/main/java/com/tutorhub/repository/SubmoduleRepository.java

```diff
@@ -2,5 +2,26 @@
 
 import com.tutorhub.model.course.Submodule;
 import org.springframework.data.jpa.repository.JpaRepository;
+import org.springframework.data.jpa.repository.Query;
 
-public interface SubmoduleRepository extends JpaRepository<Submodule, Long> {}
+import java.util.Optional;
+
+public interface SubmoduleRepository extends JpaRepository<Submodule, Long> {
+  @Query(
+      value =
+          """
+              SELECT s.id,
+                     s.name,
+                     s.module_id,
+                     s.content_id
+              FROM submodule s LEFT JOIN module m ON s.module_id = m.id
+              LEFT JOIN course c ON m.course_id = c.id
+              WHERE s.id = :submoduleId
+              AND m.id = :moduleId
+              AND c.id = :courseId
+              """,
+      nativeQuery = true)
+  Optional<Submodule> findByCourseIdAndModuleIdAndSubmoduleId(Long courseId,
```

Comment:
Did you try to use this method without query? JPA allows to build such queries itself without custom scripting

Reply:
Wow, I found the solution, thank you